### PR TITLE
validate and commit index definition

### PR DESCRIPTION
**Pull request by cendhu**
This commit validates DB index defined for the database. An index
for a database can be defined or updated for an existing database and
also can be defined when the db is created

Signed-off-by: senthil <snatara7@in.ibm.com>

### CODE REVIEW

File: internal/blockprocessor/committer.go [hunk 1]
```diff
@@ -328,24 +332,64 @@ func constructDBEntriesForDataTx(tx *types.DataTx, version *types.Version) []*wo
...
-func constructDBEntriesForDBAdminTx(tx *types.DBAdministrationTx, version *types.Version) *worldstate.DBUpdates {
+func constructDBEntriesForDBAdminTx(tx *types.DBAdministrationTx, version *types.Version) ([]*worldstate.DBUpdates, error) {
 	var toCreateDBs []*worldstate.KVWithMetadata
+	var indexForExistingDBs []*worldstate.KVWithMetadata
 
 	for _, dbName := range tx.CreateDBs {
+		var value []byte
+		dbIndex, ok := tx.DBsIndex[dbName]
+		if ok {
+			v, err := json.Marshal(dbIndex.AttributeAndType)
```
Comment (tock-ibm) [inline internal/blockprocessor/committer.go:343]: The use of `dbIndex.GetAttributeAndType()` as below is more idiomatic, no?

File: internal/blockprocessor/dbadmin_tx_validator.go [hunk 2]
```diff
@@ -147,3 +152,48 @@ func (v *dbAdminTxValidator) validateDeleteDBEntries(toDeleteDBs []string) *type
...
+	}
+	for _, dbName := range toDeleteDBs {
+		toDeleteDBsLookup[dbName] = true
+	}
+
+	for dbName, dbIndex := range dbsIndex {
+		if !v.db.Exist(dbName) && !toCreateDBsLookup[dbName] {
+			return &types.ValidationInfo{
+				Flag:            types.Flag_INVALID_INCORRECT_ENTRIES,
+				ReasonIfInvalid: "index definion provided for database [" + dbName + "] cannot be processed as the database neither exist nor in the create DB list",
```
Comment (tock-ibm) [inline internal/blockprocessor/dbadmin_tx_validator.go:171]: neither exists nor is in

**Comment by tock-ibm** [comment]
Looks good, except for couple of nits

File: internal/blockprocessor/committer.go (hunk 1 above)
Comment (cendhu) [reply internal/blockprocessor/committer.go]: Done

File: internal/blockprocessor/dbadmin_tx_validator.go (hunk 2 above)
Comment (cendhu) [reply internal/blockprocessor/dbadmin_tx_validator.go]: Done

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**